openmetadata-service/src/main/java/org/openmetadata/service/resources/search/ApplyOMDRulesForElasticSearch.java

```diff
@@ -0,0 +1,234 @@
+          if (rule.getCondition() == null) {
+            if (rule.getEffect().equals(Rule.Effect.DENY)) {
+              baseQueryBuilder.mustNot(searchSourceBuilder.query()); // 'MustNot' because of DENY rule
+            } else {
+              baseQueryBuilder.should(searchSourceBuilder.query()); // 'should' because of Allow rule
+            }
+          } else {
+            SpelExpression ruleExpression = (SpelExpression) rule.getExpression();
+            var rootAst = ruleExpression.getAST();
+            BoolQueryBuilder boolQueryBuilder = createQueryBuilderBasedOnRule(rootAst);
+            if (rule.getEffect().equals(Rule.Effect.DENY)) {
+              baseQueryBuilder.mustNot(boolQueryBuilder); // 'MustNot' because of DENY rule
+            } else {
+              baseQueryBuilder.should(boolQueryBuilder); // 'should' because of Allow rule
+            }
+          }
+          anyRuleApplied = true;
+        }
+      } // End of for
+    } // End of while
+
+    // If there is no rule applied to query, we return an empty result
+    if (!anyRuleApplied) {
+      return searchSourceBuilder.query(
+          QueryBuilders.boolQuery().must(searchSourceBuilder.query()).must(QueryBuilders.queryStringQuery("")));
+    }
+
+    /*
+        // For Apply just 'Deny' Rules
+        while (policies.hasNext()) {
+          SubjectContext.PolicyContext context = policies.next();
+          for (CompiledRule rule : context.getRules()) {
+            if ((rule.getResources().contains(indexToRuleMap.get(index))
+                    || rule.getResources().toString().toLowerCase().contains("all"))
+                    && (rule.getOperations().contains(MetadataOperation.ALL)
+                    || rule.getOperations().contains(MetadataOperation.VIEW_ALL))
+                    && rule.getEffect().equals(Rule.Effect.DENY)) {
+              // Check Conditions
+              if (rule.getCondition() == null) {
+                baseQueryBuilder.mustNot(searchSourceBuilder.query()); // 'MustNot' because of DENY rule
+              } else {
+                SpelExpression ruleExpression = (SpelExpression) rule.getExpression();
+                var rootAst = ruleExpression.getAST();
+                BoolQueryBuilder boolQueryBuilder = createQueryBuilderBasedOnRule(rootAst);
+                baseQueryBuilder.mustNot(boolQueryBuilder); // 'MustNot' because of DENY rule
+              }
+            }
+          } // End of for
+        } // End of while
+    */
+
+    // Combine searchSourceBuilder.query() with new Query that enforce policy rules
+    searchSourceBuilder.query(QueryBuilders.boolQuery().must(searchSourceBuilder.query()).must(baseQueryBuilder));
+    return searchSourceBuilder;
+  }
+
+  private BoolQueryBuilder createQueryBuilderBasedOnRule(SpelNode ast) throws IOException {
+    try {
+      BoolQueryBuilder boolQueryBuilder = QueryBuilders.boolQuery();
+      var keyword = ast2keyword(ast);
+
+      if (ast.getChildCount() == 0) {
```

Comment:
The complexity involved enforcing these on the search side is
1. Hierarchy of Rule processing
2. We process Deny rules first before the allow rules
3. Enforcing the Resource side attribute conditions example Allow Edit only if the table contains certain tags